ops/testing.py

```diff
@@ -38,8 +41,144 @@
     storage,
 )
 
-# The Harness testing framework.
-_ = ActionFailed
+# If the 'ops.testing' optional extra is installed, make those
+# names available in this namespace.
+try:
+    _version = _get_package_version('ops-scenario')
+except _PackageNotFoundError:
+    from ops._private.harness import ActionFailed  # type: ignore
+else:
+    if _version and int(_version.split('.', 1)[0]) >= 7:
+        from scenario import (
```

Comment:
🤔 don't we want to hide the name "scenario" from like everywhere?

Reply:
> 🤔 don't we want to hide the name "scenario" from like everywhere?

I think we want (or are expected to) hide it so that it's not used in charms, in our docs, etc. Essentially: you can build charms and write tests for them without ever knowing the name "Scenario".

It's not 100% hidden, e.g. you can do this:

```python
>>> repr(ops.testing.Context)
"<class 'scenario.context.Context'>"
```

Although arguably `Context.__repr__` could be set to something better. But there's still things like:

```python
>> ops.testing.State.__module__
'scenario.state'
```

I don't feel that the "I want to reserve the name Scenario" comment means that we need to entirely wipe all traces of it.